src/validate.ts

```diff
@@ -26,18 +35,64 @@ function responseValidation(apiResponse: object, schema: object) {
 
 /**
  * Identifies the response type and validates the function based on the schemas.
- * @param apiResponse User defined api response
- * @param responseType User defined intent response
+ * @param req Request from one of the generate request functions.
+ * @param apiResponse User defined api response.
  * @return Errors from AJV validation, if any. Undefined otherwise.
  */
 export function validate(intentRequest: object, apiResponse: object) {
   const responseType = intentRequest['inputs'][0]['intent'];
   if (responseType === 'action.devices.SYNC') {
-    return responseValidation(apiResponse, SYNC_RESPONSE_SCHEMA);
+    // validate traits with attributes schema
```

Comment:
I would suggest a different approach here: "iff response intent schema is valid, inspect and validate attributes"
Currently, this code assumes a lot about the response before testing againt SYNC_RESPONSE_SCHEMA which wouldn't be true if I gave you a pile of hot garbage JSON that doesn't even resemble a SYNC intent.